worker_test.go

```diff
@@ -2,21 +2,380 @@ package main
 
 import (
 	"context"
+	"crypto/rand"
+	"crypto/rsa"
 	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"math/big"
+	"net/http/httptest"
+	"os"
 	"reflect"
 	"sync"
 	"testing"
 	"time"
 
+	"github.com/aws/aws-sdk-go/service/autoscaling"
 	cloudformation "github.com/mweagle/go-cloudformation"
+	log "github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"github.com/zalando-incubator/kube-ingress-aws-controller/aws"
+	awsAdapter "github.com/zalando-incubator/kube-ingress-aws-controller/aws"
 	"github.com/zalando-incubator/kube-ingress-aws-controller/certs"
 	"github.com/zalando-incubator/kube-ingress-aws-controller/kubernetes"
+	"github.com/zalando/skipper/dataclients/kubernetes/kubernetestest"
 	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/zalando-incubator/kube-ingress-aws-controller/aws/fake"
 )
 
+// TODO(LThiesen): This should be extracted to another file.
+// This is also just a copy of certs test so it might be useful
+// to just distribute it in a separate package like aws/fake
+type mockedCertificateProvider struct {
+	t *testing.T
+}
+
+func (m mockedCertificateProvider) GetCertificates() ([]*certs.CertificateSummary, error) {
+	//tenYears := time.Hour * 24 * 365 * 10
+	/*caCert := x509.Certificate{
+		SerialNumber: big.NewInt(123),
+		Subject: pkix.Name{
+			Organization: []string{"Testing CA"},
+			CommonName:   "foo.bar.org",
+		},
+		NotBefore: time.Time{},
+		NotAfter:  time.Now().Add(tenYears),
+
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+
+		DNSNames: []string{"foo.bar.org"},
+		IsCA:     false,
+	}*/
+
+	/*caCert := x509.Certificate{
+		SerialNumber: big.NewInt(123),
+		DNSNames:     []string{"foo.bar.org"},
+		NotBefore:    time.Now(),
+		NotAfter:     time.Now().Add(time.Hour * 24),
+
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+
+		IsCA: true,
+	}*/
```

Comment:
I guess we don't need commented code. In case you want to provide an example you can create an Example func, see also golang docs about that, in case you wanted to provide an example

Reply:
Yeah, I'm sorry about that, I ended up missing these comments when preparing the code for review.